src/main/java/bio/terra/service/snapshot/flight/create/SnapshotAuthzFileAclStep.java

```diff
@@ -73,7 +73,8 @@ public StepResult doStep(FlightContext context) throws InterruptedException {
             // Now, how to figure out if the failure is due to IAM propagation delay. We know it will
             // be a 400 - bad request and the docs indicate the reason will be "badRequest". So for now
             // we will log alot and retry on that.
-            if (ex.getCode() == 400 && StringUtils.equals(ex.getReason(), "badRequest")) {
+            if (ex.getCode() == 400 && (StringUtils.equals(ex.getReason(), "badRequest") ||
+                 ex.getMessage().contains("Could not find group"))) {
```

Comment:
Do you know what causes this error? It looks a lot like the one where a user gets added to a policy that doesn't exist (e.g. adding a custodian to a snapshot), which is a real 400 error.

Reply:
I think it's the reason described in the comment above the line of code - IAM propagation delay. 

Since this step is only used on snapshot create, all of the policies should exist.

Reply:
Got it, thanks for clarifying!